### PR TITLE
✨Set default KubeadmConfig format to cloud-config✨

### LINKED ISSUE #5823: Set default value for bootstrap secret format to cloud-init
**Detailed Description**

As discussed in [#4172](https://github.com/kubernetes-sigs/cluster-api/pull/4172/files#r764995622), right now we set `format` field in bootstrap secret to empty value. We should set it via webhook to `cloud-init` explicitly.

**Anything else you would like to add:**

[Miscellaneous information that will assist in solving the issue.]

/kind feature

### PR DESCRIPTION
**Pull request by invidian**
**What this PR does / why we need it**:
So we do not produce format key in bootstrap data with empty value.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5823

NOTE: I only unit-tested this. Let me know if I should write more automated tests for it somewhere.

### CODE REVIEW

File: bootstrap/kubeadm/api/v1beta1/kubeadmconfig_webhook_test.go [hunk 1]
```diff
@@ -310,3 +310,18 @@ func TestClusterValidate(t *testing.T) {
...
+
+func TestKubeadmConfigDefault(t *testing.T) {
+	g := NewWithT(t)
+
+	kc := &KubeadmConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: metav1.NamespaceDefault,
+		},
+	}
```
Comment (fabriziopandini) [inline bootstrap/kubeadm/api/v1beta1/kubeadmconfig_webhook_test.go:322]: What about changing this in a table test so we verify both that default is applied if the values is not set and also if other values are preserved if provided.

Also, let's make sure to call

t.Run("for KubeadmConfig", utildefaulting.DefaultValidateTest(kc))

So we establish a signal that validation and defaulting do not conflict each other.
Same for KubeadmConfigTemplate

Comment (invidian) [reply bootstrap/kubeadm/api/v1beta1/kubeadmconfig_webhook_test.go:322]: Thanks for the hints. I just noticed that tests are actually failing. I'm still not very familiar with the conventions.

File: bootstrap/kubeadm/api/v1beta1/kubeadmconfig_webhook.go [hunk 2]
```diff
@@ -43,9 +43,24 @@ func (c *KubeadmConfig) SetupWebhookWithManager(mgr ctrl.Manager) error {
...
+// Default implements webhook.Defaulter so a webhook will be registered for the type.
+func (c *KubeadmConfig) Default() {
+	defaultKubeadmConfigSpec(&c.Spec)
+}
+
+func defaultKubeadmConfigSpec(s *KubeadmConfigSpec) {
+	if s.Format == "" {
+		s.Format = CloudConfig
+	}
+}
```
Comment (vincepri) [inline bootstrap/kubeadm/api/v1beta1/kubeadmconfig_webhook.go:63]: Could we default this field in the openapi schema spec rather than creating a webhook?

Comment (invidian) [reply bootstrap/kubeadm/api/v1beta1/kubeadmconfig_webhook.go]: Done, I think. Please have a look.

**Review by fabriziopandini — COMMENTED** [review]
I personally prefer to keep all the defaulting rules in one single point (the code), but the current approach is fine as well

/lgtm

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**